teku/src/main/java/tech/pegasys/teku/cli/subcommand/debug/DebugDbCommand.java

```diff
@@ -439,6 +447,290 @@ public int dumpBlobsSidecar(
+                database
+                    .getHotBlock(blobsSidecarsKey.getBlockRoot())
+                    .or(
+                        () ->
+                            database
+                                .getFinalizedBlockAtSlot(blobsSidecarsKey.getSlot())
+                                .filter(
+                                    block ->
+                                        block.getRoot().equals(blobsSidecarsKey.getBlockRoot())));
+
+            if (maybeBlockFound.isEmpty()) {
+              System.err.printf(
+                  "ERROR: confirmed blobs sidecar has no corresponding finalized or hot block %s%n",
+                  blobsSidecarsKey.toLogString());
+            }
+          }
+
+          // check Data Availability
+          if (blobsSidecarsKey.getSlot().isLessThan(lowerSlotBoundDataAvailability)) {
+            unprunedBlobsSidecars++;
+          }
+        }
+      }
+
+      // check that unconfirmed blobs sidecar exists in the main blobs sidecar table
+      System.out.printf("Checking unconfirmed blobs sidecar in main blobs sidecar table...%n");
+
+      for (final SlotAndBlockRoot slotAndBlockRoot : unconfirmedBlobsSidecars) {
+        if (maybeLatestBlobsSidecarSlot
+            .map(slot -> slotAndBlockRoot.getSlot().isGreaterThan(slot))
+            .orElse(false)) {
+          // unconfirmed sidecars for a non finalized slot
+          unprunedUnconfirmedBlobsSidecarsNonFinalized++;
+        } else {
+          // unconfirmed sidecars for finalized slot
+          unprunedUnconfirmedBlobsSidecarsFinalized++;
+        }
+
+        if (database.getBlobsSidecar(slotAndBlockRoot).isEmpty()) {
+          System.err.printf(
+              "ERROR: Unable to locate unconfirmed blobs sidecar in main blobs sidecar table %s%n",
+              slotAndBlockRoot.toLogString());
+        }
+      }
+    } catch (DatabaseStorageException ex) {
+      System.out.println("Failed to open database");
+    }
+
+    final long endTime = System.currentTimeMillis();
+    System.out.printf(
+        "Done. Performed %s block->blobs_sidecars and %s blobs_sidecars->block checks in %s ms%n",
+        blocksToSidecarsCounter, sidecarsToBlocksCounter, endTime - startTimeMillis);
+    System.out.printf(
+        "\tCalculated Data Availability window (slots): %s - %s%n",
+        lowerSlotBoundDataAvailability, currentSlot);
+    System.out.printf("\tEarliest BlobsSidecar slot: %s%n", maybeEarliestBlobsSidecarSlot);
+    System.out.printf(
+        "\tEarliest finalized block slot: %s%n",
+        maybeEarliestBlock.map(SignedBeaconBlock::getSlot));
+    System.out.printf("\tLatest finalized block slot: %s%n", maybeLatestBlobsSidecarSlot);
+    System.out.printf("\tMissing finalized BlobsSidecars: %s%n", missingFinalizedBlobsSidecars);
+    System.out.printf("\tMissing hot BlobsSidecars: %s%n", missingHotBlobsSidecars);
+    System.out.printf("\tExcess BlobsSidecars: %s%n", blobsSidecarsWithoutBlocks);
```

Comment:
Ok, I see, it's some total of unpruned from other side. I think it's just needs better name